text/0079-exception-groups.md

```diff
@@ -0,0 +1,348 @@
+- Start Date: 2023-03-06
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/79
+- RFC Status: draft
+
+# Summary
+
+This RFC is intended to capture the changes required to Sentry to handle *exception groups*,
+including changes to SDKs, protocol schema, ingestion, and UI.
+
+# Motivation
+
+Several programming languages have a concept of an unrelated group of multiple exception, aggregated into a single exception.
+
+- Python [`exception_group`](https://docs.python.org/3/library/exceptions.html#exception-groups) ([PEP 654](https://peps.python.org/pep-0654/))
+- .NET [`AggregateException`](https://docs.microsoft.com/dotnet/api/system.aggregateexception)
+- JavaScript [`AggregateError`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/AggregateError)
+- Go [`Unwrap() []error`](https://pkg.go.dev/errors) ([see also](https://tip.golang.org/doc/go1.20#errors))
+- Java [`Throwable.getSuppressed()`](https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getSuppressed())
+
+There may be others. We will use the term "exception group" throughout the design.
+It applies to the concept, regardless of language.
+
+Sentry needs a way to capture exception groups, and present them in a meaningful way.
+Simply raising the exception group as a single issue is insufficient.
+
+# Background
+
+## About Exception Groups
+
+An exception group is an exception unto itself.  It is caused by the exceptions in the group,
+but there is no implied causal relationship between those exceptions.
```

Comment:
If exceptions in a group have no causal relationship, the exceptions in the group are likely to be viewed as separate issues in Sentry. In this case, is there some benefit we would gain by reporting them together and showing the connection? This could be elaborated on in the intro.

Reply:
I experimented with sending separate events, and it's not a viable option.  I'll add details about it to the RFC though, so that it's understood.  Thanks.

Reply:
The RFC has been changed considerably.  The connection between exceptions in an exception group is only important with regard to the exception group tree itself.  Linking separate issues together in Sentry is less important.  Thanks.